tenseal/tensors/ckksvector.cpp

```diff
@@ -85,7 +85,15 @@ CKKSVector& CKKSVector::add_inplace(CKKSVector to_add) {
     }
 
     if (this->size() != to_add.size()) {
-        throw invalid_argument("can't add vectors of different sizes");
+        if (this->size() == 1) {
+            this->replicate_first_slot_inplace(to_add.size());
+            return this->add_inplace(to_add);
+        } else if (to_add.size() == 1) {
+            CKKSVector replicated = to_add.replicate_first_slot(this->size());
+            return this->add_inplace(replicated);
```

Comment:
why not just continue instead of making one more call?

Reply:
if you are talking about the 2nd check, the multiplication could be either Nx1 or 1xN, so we need also to check if the 2nd vector is of size 1.